test/e2e/upgrade_test.go

```diff
@@ -31,15 +31,70 @@ const (
 	windowsWorkloadTesterJob = "windows-workload-tester"
 )
 
+// getNodeAffinities generate node affinity for non-BYOH and BYOH nodes if present
+func getNodeAffinities() ([]*v1.Affinity, error) {
```

Comment:
This name made me think the function was returning a slice of affinities for each individual node, how about `getNodeTypeAffinities()` instead? Or something like that. Just something a little more specific :)